### DISCUSSION TITLE
How to embed yaegi in my C/C++ application?

### DISCUSSION THREAD

**Discussion by vit1251** [traefik/yaegi#1272] [Q&A]
I have a strong set C API and want to provide it in Golang source code. But some my API non-thread safe. Could you provide any idea how to synchronize C API calls and prevent Golang native (OS) invoke thread switching?

**Comment by cauefcr** [comment]
There is LockOSThread() for preveting golang from swithing the goroutine to another thread.

**Comment by vit1251** [comment]
OK. How about embedding code in C/C++ ?

**Comment by JAicewizard** [comment]
https://medium.com/learning-the-go-programming-language/calling-go-functions-from-other-languages-4c7d8bcc69bf
